source/simulator/solver_schemes.cc

```diff
@@ -214,6 +215,14 @@ namespace aspect
           }
       }
 
+    // Update the copy fields *after* we have solved all the other fields
+    for (unsigned int c=0; c < introspection.n_compositional_fields; ++c)
+      {
+        const AdvectionField adv_field (AdvectionField::composition(c));
+        if (adv_field.advection_method(introspection) == Parameters<dim>::AdvectionFieldMethod::prescribed_field)
+          interpolate_material_output_into_field();
```

Comment:
this looks wrong. I think you should call this method only once even if you have several prescribed fields.

Reply:
Yes, you're right.